src/woo_publications/publications/models.py

```diff
@@ -133,6 +138,43 @@ def get_owner(self) -> ActingUser | None:
         assert isinstance(log, TimelineLogProxy)
         return log.acting_user[0]
 
+    def revoke_own_published_documents(
+        self, user: User | ApiUser, remarks: str | None = None
+    ) -> None:
+        documents = (
+            self.document_set.filter(  # pyright: ignore reportAttributeAccessIssue
+                publicatiestatus=PublicationStatusOptions.published
+            )
+        )
+
+        match user:
+            case User():
+                for document in documents:
+                    document.publicatiestatus = PublicationStatusOptions.revoked
+                    document.laatst_gewijzigd_datum = timezone.now()
+                    audit_admin_update(
+                        content_object=document,
+                        django_user=user,
+                        object_data=serialize_instance(document),
+                    )
+            case dict():
+                assert remarks
+                for document in documents:
+                    document.publicatiestatus = PublicationStatusOptions.revoked
+                    document.laatst_gewijzigd_datum = timezone.now()
+                    audit_api_update(
+                        content_object=document,
+                        user_id=user["user_id"],
+                        user_display=user["user_repr"],
+                        object_data=serialize_instance(document),
+                        remarks=remarks,
+                    )
+
+        # manually update laatst_gewijzigd_datum because bulk_update doesn't trigger save method.
+        Document.objects.bulk_update(
+            documents, ["publicatiestatus", "laatst_gewijzigd_datum"]
+        )
```

Comment:
if the `auto_now` isn't taken into account automatically, you can still do `published_documents.update(..., laatst_gewijzigd_datum=timezone.now())`